prow/github/client.go

```diff
@@ -336,36 +336,57 @@ func (c *Client) CreateIssueReaction(org, repo string, ID int, reaction string)
 	return err
 }
 
-// ListIssueComments returns all comments on an issue. This may use more than
-// one API token.
-func (c *Client) ListIssueComments(org, repo string, number int) ([]IssueComment, error) {
-	c.log("ListIssueComments", org, repo, number)
-	if c.fake {
-		return nil, nil
-	}
-	nextURL := fmt.Sprintf("%s/repos/%s/%s/issues/%d/comments?per_page=100", c.base, org, repo, number)
-	var comments []IssueComment
-	for nextURL != "" {
-		resp, err := c.requestRetry(http.MethodGet, nextURL, "", nil)
+type marshallingCallback func(body []byte) error
+
+// readPaginatedResults iterates over all objects in the paginated
+// result indicated by the given url.  The provided callback is called
+// for each object.  An error is returned if encountered in making
+// calls to github or marshalling objects.
+func (c *Client) readPaginatedResults(path string, callback marshallingCallback) error {
```

Comment:
It looks like all the users have effectively identical marshallingCallback functions, except for the types they're unmarshaling.

Could you change the function signature to `readPaginatedResults(path string, results []interface{})`, and accumulate the results into the provided slice?